sympy/functions/combinatorial/factorials.py

```diff
@@ -782,23 +782,24 @@ class binomial(CombinatorialFunction):
     .. math:: \binom{n}{k} = \frac{n!}{k!(n-k)!}\ \text{or}\
                 \binom{n}{k} = \frac{ff(n, k)}{k!}
 
-    First, in a strict combinatorial sense it defines the
+    First, in a strict combinatorial sense, it defines the
     number of ways we can choose `k` elements from a set of
     `n` elements. In this case both arguments are nonnegative
     integers and binomial is computed using an efficient
     algorithm based on prime factorization.
 
-    The other definition is generalization for arbitrary `n`,
-    however `k` must also be nonnegative. This case is very
-    useful when evaluating summations.
+    The other definition is a generalization for an arbitrary `n`,
+    and nonnegative `k`. This case is very useful when evaluating
+    summations.
 
-    For the sake of convenience for negative integer `k` this function
-    will return zero no matter what valued is the other argument.
+    For the sake of convenience, this function will return zero
+    for any negative integer `k`, except cases where `n` equals `k`,
+    in which instance the function returns one.
```

Comment:
```python
>>> binomial(x,x)
1
>>> binomial(-1,-1)
0
```

Reply:
https://github.com/sympy/sympy/pull/18960#discussion_r403443020
we need to fix that right ?